src/wasm-lib/kcl/src/settings/types/mod.rs

```diff
@@ -280,6 +284,57 @@ pub struct ModelingSettings {
     pub show_scale_grid: bool,
 }
 
+fn named_view_point_version_one() -> f64 {
+    1.0
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema, ts_rs::TS, Validate)]
+#[serde(rename_all = "snake_case")]
+#[ts(export)]
+pub struct NamedView {
+    /// User defined name to identify the named view. A label.
+    #[serde(default, alias = "name", skip_serializing_if = "is_default")]
+    pub name: String,
+    /// Engine camera eye off set
+    #[serde(default, alias = "eyeOffset", skip_serializing_if = "is_default")]
+    pub eye_offset: f64,
+    /// Engine camera vertical FOV
+    #[serde(default, alias = "fovY", skip_serializing_if = "is_default")]
+    pub fov_y: f64,
+    // Engine camera is orthographic or perspective projection
+    #[serde(default, alias = "isOrtho")]
+    pub is_ortho: bool,
+    /// Engine camera is orthographic camera scaling enabled
+    #[serde(default, alias = "orthoScaleEnabled")]
+    pub ortho_scale_enabled: bool,
+    /// Engine camera orthographic scaling factor
+    #[serde(default, alias = "orthoScaleFactor", skip_serializing_if = "is_default")]
+    pub ortho_scale_factor: f64,
+    /// Engine camera position that the camera pivots around
+    #[serde(default, alias = "pivotPosition", skip_serializing_if = "is_default")]
+    pub pivot_position: [f64; 3],
+    /// Engine camera orientation in relation to the pivot position
+    #[serde(default, alias = "pivotRotation", skip_serializing_if = "is_default")]
+    pub pivot_rotation: [f64; 4],
+    /// Engine camera world coordinate system orientation
+    #[serde(default, alias = "worldCoordSystem", skip_serializing_if = "is_default")]
+    pub world_coord_system: String,
+    /// A unique uuidv4 to have a unique key when storing named views
+    #[serde(default = "uuid::Uuid::new_v4")]
+    pub id: uuid::Uuid,
+    /// Version number of the view point if the engine camera API changes
+    #[serde(default = "named_view_point_version_one")]
+    pub version: f64,
+}
+
+impl PartialEq for NamedView {
+    fn eq(&self, other: &Self) -> bool {
+        self.id == other.id && self.version == other.version
+    }
```

Comment:
All other fields are ignored? Why is this? It seems like this will be very confusing to the next person who works on this.

Reply:
tbh, no clue. I thought I could handle uniqueness within the serialization with this but I don't think this does anything actually. 

I wanted a check that says if you have two named views with the same id and version that they are the same, I didn't care about the camera values but this actually doesn't do anything I want.

serde will still serialize this if they match...

Reply:
Maybe a hash comparison is better?

Reply:
If I understood correctly, the conclusion from the Slack thread was to change from an array to a map keyed by the ID.

Reply:
This resolved now.